tools/engine_tool/lib/src/commands/format_command.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io' as io;
+
+import 'package:path/path.dart' as p;
+
+import '../logger.dart';
+import 'command.dart';
+import 'flags.dart';
+
+// The format command implementation below works by spawning another Dart VM to
+// run the program under ci/bin/format.dart. Instead, format.dart should be
+// moved under the engine_tool package and invoked by a function call. The file
+// ci/bin/format.dart should be split up so that each of its `FormatCheckers` is
+// in a seprate file under src/formatters, and they should be unit-tested.
+
+/// The 'format' command.
+final class FormatCommand extends CommandBase {
+  /// Constructs the 'query' command.
```

Comment:
This comment and the comment above on the class declaration aren't particularly helpful. I'd consider removing them. See: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-useless-documentation

```suggestion
  /// Constructs the 'format' command.
```